### DISCUSSION TITLE
Inquiry on Deepflow's Support for Large-Scale Deployments

### DISCUSSION THREAD

**Discussion by ANSANJAY** [deepflowio/deepflow#5517] [General]
Hello Deepflow Team,

As we're exploring observability solutions that align with our infrastructure's scale and complexity, I'm interested in understanding more about Deepflow's features and compatibility. 
 I have a few questions that would help in evaluating Deepflow as a potential fit:

1. **Kubernetes Integration**: Does Deepflow offer out-of-the-box support for Kubernetes environments, and if so, could you highlight how it integrates with K8s clusters?
2. **VM Monitoring**: Is Deepflow capable of monitoring virtual machine (VM) environments alongside containerized applications?
3. **VM Customization**: For deployments involving VMs, does Deepflow allow for customization to address specific monitoring needs?
4. **Ideal Use Cases**: Could you share scenarios or use cases where Deepflow particularly excels, especially in large-scale environments?
5. **Deployment Strategy**: For organizations managing a substantial fleet of VMs and Kubernetes clusters, what best practices does Deepflow recommend for deployment? Additionally, are there examples of large organizations successfully leveraging Deepflow for their observability needs?

I appreciate any insights you can provide, which will help in our decision-making process.

Thank you.

**Comment by sharang** [comment]
@ANSANJAY,

I just replied to your Gmail, sorry for the late response, looking forward to further communication.
